src/script/engines/lua.c

```diff
@@ -1032,6 +1063,39 @@ const char* _luaGetError(struct mScriptEngineContext* context) {
 	return luaContext->lastError;
 }
 
+void _luaNewMetatable(lua_State* lua, const char* name, const luaL_Reg* funcs) {
+	luaL_newmetatable(lua, name);
+#if LUA_VERSION_NUM < 502
+	lua_pushstring(lua, name);
+	lua_setfield(lua, -2, "__name");
+	luaL_register(lua, NULL, funcs);
+#else
+	luaL_setfuncs(lua, funcs, 0);
+#endif
+	lua_pop(lua, 1);
+}
+
+const char* _luaTolstring(lua_State* lua, int idx, size_t* len) {
```

Comment:
@endrift Bikeshed question: Should this function wrap `luaL_tostring` for 5.3+, or should we polyfill `luaL_tostring` so that 5.3+ code compiles under 5.1/5.2?

(Tangentially related, but not a task for this PR: Is there a particular reason why the internal functions in this file aren't declared static?)